benefits/core/admin/claims.py

```diff
@@ -0,0 +1,35 @@
+from django.contrib import admin
+
+from benefits.core import models
+from .users import is_staff_or_superuser
+
+
+@admin.register(models.ClaimsProvider)
+class ClaimsProviderAdmin(admin.ModelAdmin):
+    def get_exclude(self, request, obj=None):
+        fields = []
+
+        if not is_staff_or_superuser(request.user):
+            fields.extend(["authority", "scheme"])
+        if not request.user.is_superuser:
+            fields.extend(["client_id_secret_name"])
+
+        return fields or super().get_exclude(request, obj)
+
+    def get_readonly_fields(self, request, obj=None):
+        fields = []
+
+        if not is_staff_or_superuser(request.user):
+            fields.extend(["client_name"])
+        if not request.user.is_superuser:
+            fields.extend(
+                [
+                    "sign_out_button_template",
+                    "sign_out_link_template",
+                    "client_id_secret_name",
```

Comment:
I was debugging and I think we may want to remove `client_id_secret_name` from the read only fields to match our [admin config](https://docs.google.com/spreadsheets/d/1lPp_LJTfJoWCyOkRC4RaljABGG_jPWOgm4gaAHJigMU/edit):

![image](https://github.com/user-attachments/assets/e0d4c7ce-7d42-41f7-903f-f849e4fcf3b4)